### PR TITLE
Set release branch for new releases

### PR DESCRIPTION
**Pull request by dlarocque**
We used to just automatically set the release branch to be `releases/<releasename>.release`, but now we accept the release branch from the payload.

### CODE REVIEW

**Review by davidmotson — APPROVED** [review]

File: functions/validation/validation.js [hunk 1]
```diff
@@ -140,6 +140,32 @@ function validateUniqueReleaseNames(releases) {
...
+    errors.push({
+      message: ERRORS.MISSING_RELEASE_FIELD,
+      offendingRelease: release,
+    });
+  } else if (typeof release.releaseBranchName !== "string" ||
+          release.releaseOperator.trim() === "") {
+    errors.push({
+      message: ERRORS.INVALID_RELEASE_FIELD,
+      offendingRelease: release,
+    });
```
Comment (daymxn) [inline functions/validation/validation.js:163]: On today's episode of _"fighting the linter"_:
```js
const isNotString = typeof release.releaseBranchName !== "string";
const isEmpty = release.releaseOperator.trim() === ""

if (!release.releaseBranchName) {
  errors.push({
    message: ERRORS.MISSING_RELEASE_FIELD,
    offendingRelease: release,
  });
} else if (isNotString || isEmpty) {
  errors.push({
    message: ERRORS.INVALID_RELEASE_FIELD,
    offendingRelease: release,
  });
}
```

although, it's a bit of a design choice and not a big deal. feel free to keep it as is if you feel it reads easier.

**Review by daymxn — COMMENTED** [review]
A few worth addressing, others not so much. LGTM once you fix the ones that need fixing

File: functions/test/validation/validation.test.js [hunk 2]
```diff
@@ -270,18 +301,33 @@ describe("validateNewReleasesStructure", () => {
...
+    const newReleases = [{
+      releaseName: "M100",
+      releaseOperator: "operator1",
+      codeFreezeDate: Timestamp.now(),
+      releaseDate: Timestamp.now(),
+      // missing release branch
+    }];
+    expect(() => validateNewReleasesStructure(newReleases)).
+        to.throw("Each release should have a string property"+
+        " property 'releaseBranchName'");
```
[the remark this answers is not in the document]
Comment (dlarocque) [reply functions/test/validation/validation.test.js:321]: yup

File: functions/test/validation/validation.test.js [hunk 3]
```diff
@@ -270,18 +301,33 @@ describe("validateNewReleasesStructure", () => {
...
           " property 'releaseDate'");
   });
 
+  it("should throw an error if a release is missing a release branch", () => {
+    const newReleases = [{
+      releaseName: "M100",
+      releaseOperator: "operator1",
+      codeFreezeDate: Timestamp.now(),
+      releaseDate: Timestamp.now(),
+      // missing release branch
```
[the remark this answers is not in the document]
Comment (dlarocque) [reply functions/test/validation/validation.test.js:318]: IMO the comment is easier to read at a quick glance and allows the names to be consistent across the tests with similar structures.

**Pull request merged**